src/org/graphstream/algorithm/LongestPath.java

```diff
@@ -0,0 +1,144 @@
+        if (weighted) {
+            fillDistanceMapWeighted(aSortedArray);
+        } else {
+            fillDistanceMapUnweighted(aSortedArray);
+        }
+        Map.Entry<Node, Double> maxEntry = getMaxEntryOfMap();
+        longestPathNode = maxEntry;
+        longestPath.add(maxEntry.getKey());
+        getMaxNeigbourgh(maxEntry.getKey());
+        Collections.reverse(longestPath);
+    }
+
+    private void fillDistanceMapWeighted(Node[] theSortedArray) {
+        for (Node aNode : theSortedArray) {
+            for (Edge anEdge : aNode.getEachEnteringEdge()) {
+                Node aSourceNode = anEdge.getSourceNode();
+                Node aTargetNode = anEdge.getTargetNode();
+                double aWeight = anEdge.getNumber("weight");
+                Double aMaxDistance = Math.max(distanceMap.get(aTargetNode), distanceMap.get(aSourceNode) + aWeight);
+                distanceMap.put(aTargetNode, aMaxDistance);
+            }
+        }
+    }
+
+    private void fillDistanceMapUnweighted(Node[] theSortedArray) {
+        for (Node aNode : theSortedArray) {
+            for (Edge anEdge : aNode.getEachEnteringEdge()) {
+                Node aSourceNode = anEdge.getSourceNode();
+                Node aTargetNode = anEdge.getTargetNode();
+                Double aMaxDistance = Math.max(distanceMap.get(aTargetNode), distanceMap.get(aSourceNode)) + 1;
+                distanceMap.put(aTargetNode, aMaxDistance);
+            }
+        }
+    }
+
+    private Map.Entry<Node, Double> getMaxEntryOfMap() {
+        Map.Entry<Node, Double> maxEntry = null;
+        for (Map.Entry<Node, Double> entry : distanceMap.entrySet()) {
+            if (maxEntry == null || entry.getValue().compareTo(maxEntry.getValue()) > 0) {
+                maxEntry = entry;
+            }
+        }
+        return maxEntry;
+    }
+
+    private void getMaxNeigbourgh(Node theNode) {
+        Node aMaxNode = null;
+        double aMaxDistance = 0.0;
+        for (Edge anEdge : theNode.getEachEnteringEdge()) {
+            Node aSourceNode = anEdge.getSourceNode();
+            if (distanceMap.get(aSourceNode) >= aMaxDistance) {
+                aMaxDistance = distanceMap.get(aSourceNode);
+                aMaxNode = aSourceNode;
+            }
+        }
+        if (aMaxNode != null) {
+            longestPath.add(aMaxNode);
+            getMaxNeigbourgh(aMaxNode);
+        }
+
+    }
+
+    private void initializeAlgorithm() {
+        for (Node aNode : graph.getEachNode()) {
+            for (Edge anEdge : aNode.getEachEdge()) {
+                double aWeight = anEdge.getNumber("weight");
+                if (Double.isNaN(aWeight)) {
+                    weighted = false;
+                }
+            }
+            distanceMap.put(aNode, 0.0);
+        }
+    }
+
+    /**
+     * gets sorted list of the longest path
+     * @return sorted list of nodes in longest path
+     */
+    public List<Node> getLongestPath() {
+        return longestPath;
+    }
+
```

Comment:
In addition to this one, a method returning the longest path as an `org.graphstream.graph.Path` object will be useful.

Reply:
sure I can look into that